Source/WebCore/rendering/StyledMarkedText.cpp

```diff
@@ -136,18 +210,36 @@ Vector<StyledMarkedText> StyledMarkedText::subdivideAndResolve(const Vector<Mark
         return { styledMarkedText };
     }
 
-    auto markedTexts = MarkedText::subdivide(textsToSubdivide);
+    auto markedTexts = MarkedText::subdivide(textsToSubdivide, OverlapStrategy::None);
     ASSERT(!markedTexts.isEmpty());
     if (UNLIKELY(markedTexts.isEmpty()))
         return { };
 
+    if (!markedTexts.isEmpty()) {
+        // Check if vector contains custom highlights.
+        bool containsHighlights = markedTexts.containsIf([](const auto& item) {
+            return item.type == MarkedText::Type::Highlight;
+        });
+
+        // Sort custom highlights to follow correct priority/insertion order.
+        if (containsHighlights) {
+            orderHighlights(markedTextsNames, markedTexts);
+
+            frontmostMarkedTexts.reserveInitialCapacity(markedTexts.size());
```

Comment:
I think this can be done more concisely with WTF::map():
```
frontmostMarkedTexts = WTF::map(markedTexts, [&](auto& markedTexts) {
    return resolveStyleForMarkedText(markedText, baseStyle, renderer, lineStyle, paintInfo);
});
```